configure.ac

```diff
@@ -247,6 +268,27 @@ if test "x$enable_debug" = xyes; then
     fi
 fi
 
+# Checking for -fsanitize flags works differently from other compiler flags.
+# That's because with GCC there's an implicit linking dependency on libasan,
+# libtsan, etc. when using the related -fsanitize flags. Clang works
+# differently: it statically links these dependencies. Therefore we can't use
+# AC_CHECK_LIB to test this, we need to compile and link.
+if test "x$enable_asan" = xyes; then
+  BITCOIN_CHECK_COMPILE_FLAG_AND_LINK([-fsanitize=address],
+                                      [CXXFLAGS="$CXXFLAGS -fsanitize=address"],
+                                      [AC_MSG_ERROR([-fsanitize=asan broken, check if you have libasan installed])])
+fi
+if test "x$enable_tsan" = xyes; then
+  BITCOIN_CHECK_COMPILE_FLAG_AND_LINK([-fsanitize=thread],
+                                      [CXXFLAGS="$CXXFLAGS -fsanitize=thread"],
```

Comment:
As mentioned in the debugging PR, Please add these to a SANITIZE_CXXFLAGS or DEBUG_CXXFLAGS or so, rather than CXXFLAGS directly.

Adding to CXXFLAGS means that future tests will be done with sanitizers enabled, which can cause false-negatives.

Reply:
will fix